backend/common/ai_drive.py

```diff
@@ -145,29 +144,39 @@ def dl_tabular_drive(
     return train_loss_results
 
 
-def dl_img_drive(
-    train_transform,
-    test_transform,
-    user_arch,
-    criterion,
-    optimizer_name,
-    default,
-    epochs,
-    batch_size,
-    shuffle,
-    IMAGE_UPLOAD_FOLDER,
-):
-    print(user_arch)
+def dl_img_drive(trainspace_data: TrainspaceData):
+    params = trainspace_data.parameters_data
+
+    optimizer_name = params["optimizer_name"]
+    criterion = params["criterion"]
+    default = (
+        trainspace_data.dataset_data["name"]
+        if trainspace_data.dataset_data.get("is_default_dataset")
+        else None
+    )
+    epochs = params.get("epochs", 5)
+    shuffle = params.get("shuffle", True)
+    batch_size = params.get("batch_size", 20)
+    train_transform = params["train_transform"]
+    test_transform = params["test_transform"]
+
+    user_arch = params["layers"]
+
     model = DLModel(parse_deep_user_architecture(user_arch))
 
     train_transform = parse_deep_user_architecture(train_transform)
     test_transform = parse_deep_user_architecture(test_transform)
 
     if not default:
-        for x in os.listdir(IMAGE_UPLOAD_FOLDER):
-            if x != ".gitkeep":
-                zip_file = os.path.join(os.path.abspath(IMAGE_UPLOAD_FOLDER), x)
-                break
+        uid = trainspace_data.uid
+        filename = trainspace_data.dataset_data["name"]
+        read_from_bucket(
+            FILE_UPLOAD_BUCKET_NAME,
+            f"{uid}/image/{filename}",
```

Comment:
can we think of ways to make the path to s3 object robust to new data sources. like could we generalize the behavior of `read_from_bucket` into something like 

```
read_from_bucket(FILE_UPLOAD_BUCKET_NAME, dataSource.toLowerCase(), name)
```
that way we can better abstract the business logic from the data logic